Lib/calendar.py

```diff
@@ -349,11 +349,27 @@ def formatday(self, day, weekday, width):
             s = '%2i' % day             # right-align single-digit days
         return s.center(width)
 
-    def formatweek(self, theweek, width):
+    def formatweek(self, theweek, width, *, highlight_day=None):
         """
         Returns a single week in a string (no newline).
         """
-        return ' '.join(self.formatday(d, wd, width) for (d, wd) in theweek)
+        if highlight_day:
+            from _colorize import get_colors
+
+            ansi = get_colors()
+            highlight = f"{ansi.BLACK}{ansi.BACKGROUND_YELLOW}"
+            reset = ansi.RESET
+        else:
+            highlight = reset = ""
+
+        return ' '.join(
+            (
+                f"{highlight}{self.formatday(d, wd, width)}{reset}"
+                if d == highlight_day
+                else self.formatday(d, wd, width)
+            )
+            for (d, wd) in theweek
+        )
```

Comment:
This is an interesting approach, why not do it in `formatday` itself?

Reply:
Ok, good to know. What's wrong with passing the parameter down? I suspect that the implementation would be significantly simpler if `formatday` directly handled highlighting.

Reply:
One reason is to keep the diff smaller, avoid changing things that don't necessarily need changing (Chesterton's fence).

Moving it down could like this (excluding docs):

```diff
diff --git a/Lib/calendar.py b/Lib/calendar.py
index d2e5e08d02d..595c90cb186 100644
--- a/Lib/calendar.py
+++ b/Lib/calendar.py
@@ -339,35 +339,29 @@ def prweek(self, theweek, width):
         """
         print(self.formatweek(theweek, width), end='')
 
-    def formatday(self, day, weekday, width):
+    def formatday(self, day, weekday, width, *, highlight_day=None):
         """
         Returns a formatted day.
         """
         if day == 0:
             s = ''
+        elif day == highlight_day:
+            from _colorize import get_colors
+
+            ansi = get_colors()
+            highlight = f"{ansi.BLACK}{ansi.BACKGROUND_YELLOW}"
+            reset = ansi.RESET
+            s = f"{highlight}{day:2}{reset}"  # right-align single-digit days
         else:
-            s = '%2i' % day             # right-align single-digit days
+            s = f"{day:2}"  # right-align single-digit days
         return s.center(width)
 
     def formatweek(self, theweek, width, *, highlight_day=None):
         """
         Returns a single week in a string (no newline).
         """
-        if highlight_day:
-            from _colorize import get_colors
-
-            ansi = get_colors()
-            highlight = f"{ansi.BLACK}{ansi.BACKGROUND_YELLOW}"
-            reset = ansi.RESET
-        else:
-            highlight = reset = ""
-
         return ' '.join(
-            (
-                f"{highlight}{self.formatday(d, wd, width)}{reset}"
-                if d == highlight_day
-                else self.formatday(d, wd, width)
-            )
+            (self.formatday(d, wd, width, highlight_day=highlight_day))
             for (d, wd) in theweek
         )
 
```

Reply:
I don't really have a strong preference.

I think most of these formatting methods are essentially for internal use, and if it was written today and not ~20 years I reckon they would be. 

I'm not sure how useful it really useful for users to call `formatweek([(9, 0), (10, 1), (11, 2), (12, 3), (13, 4), (14, 5), (15, 6)], 2)` and get:

```
 9 10 11 12 13 14 15
```


Or call `formatday(9, 0, 2)` get:

```
 9
```

@erlend-aasland What do you think?

Reply:
I agree with Hugo.